src/usecase_regression.Rmd

```diff
@@ -0,0 +1,107 @@
+
+```{r}
+set.seed(1234)
+
+## Define the learners for the tuning
+base.learners = list(
+  makeLearner("regr.ksvm"),
+  makeLearner("regr.ranger"),
+  makeLearner("regr.penalized.lasso")
+)
+lrn = makeModelMultiplexer(base.learners)
+
+## Define a search space for each parameter
+ps = makeModelMultiplexerParamSet(lrn,
+  makeNumericParam("sigma", lower = -12, upper = 12, trafo = function(x) 2^x),
+  makeIntegerParam("num.trees", lower = 1L, upper = 700L),
+  makeNumericParam("lambda1",  lower = 0, upper = 20)
+  
+)
+
+## Choose a resampling strategy
+rdesc = makeResampleDesc("CV", iters = 10L)
+
+## Choose a performance measure
+meas = rmse
+
+## Choose a tuning method
+ctrl = makeTuneControlRandom(maxit = 300L)
+
+## Tune the parameters
+res = tuneParams(lrn, task = regr.task, resampling = rdesc, par.set = ps,
+                                measures = meas, control = ctrl, show.info = FALSE)
+## Extract the results
+res.df = as.data.frame(trafoOptPath(res$opt.path))
+head(res.df)
+
+## Extract the optimal parameter values for each learner
+res.lasso = res.df[(res.df$selected.learner=="regr.penalized.lasso"), c("regr.penalized.lasso.lambda1", "rmse.test.rmse")]
+lambda1 = list("lambda1" = res.lasso$regr.penalized.lasso.lambda1[which.min(res.lasso[, 2])])
+
+res.svm = res.df[(res.df$selected.learner=="regr.ksvm"), c("regr.ksvm.sigma", "rmse.test.rmse")]
+sigma = list("sigma" = res.svm$regr.ksvm.sigma[which.min(res.svm[, 2])])
+
+res.rf = res.df[(res.df$selected.learner=="regr.ranger"), c("regr.ranger.num.trees", "rmse.test.rmse")]
+num.trees = list("num.trees" = res.rf$regr.ranger.num.trees[which.min(res.rf[, 2])])
+```
+Finally, we extracted the optimal parameter value for each learner. Now we are ready to proceed with the benchmark experiment!
+
+### Benchmark Experiment
+In order to conduct a [benchmark experiment](benchmark_experiments.md), it is necessary to choose an evaluation method. We will use the resampling strategy from the previous section and two performance measures - root mean squared error (``rmse``) and mean of absolute error (``mae``).
+
+```{r}
+## Four learners to be compared
+lrns = list(makeLearner("regr.lm"), 
+            makeLearner("regr.ksvm", par.vals = sigma),
+            makeLearner("regr.ranger", par.vals = num.trees),
+            makeLearner("regr.penalized.lasso", par.vals = lambda1))
+
+## Conduct the benchmark experiment
+bmr = benchmark(lrns, regr.task, rdesc, measures = list(rmse, mae), show.info = FALSE)
+```
+
+
+### Performance
+
+Now we want to evaluate the results.
+```{r}
+head(getBMRPerformances(bmr, as.df = TRUE))
+
+```
+
+A closer look at the first boxplot reveals that random forest outperforms the other learners for this specific task. However, the SVM model using the measure ``mae`` seems to perform better than the other models. Moreover, in both cases the benchmark experiment for the linear regression and lasso regression models yield similar but poor results.
```

Comment:
Not sure if "Moreover" is suitable in this context. Something like this: (?)
"In both cases, the benchmark experiments for linear and lasso regression models yield similar but poor results."